src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -2044,9 +2044,11 @@ private boolean checkReFetch() {
 		if ((currentFirstCursor < 0)
 		        || (cursorPosition >= 0 && ((cursorPosition < currentFirstCursor) || (cursorPosition > currentFirstCursor
 		                + fetchedTupleNumber - 1)))) {
-			fetch();
-			if (errorHandler.getErrorCode() != UErrorCode.ER_NO_ERROR)
-				return false;
+			if (tuples[cursorPosition] == null) {
```

Comment:
I don't know exactly this routine. However, one concern is whether it is correct not to fetch again if tuples[cursorPosition]  is not null in updatable cursor.

Reply:
for updatable cursor, the fetch is processed unconditionally through reFech( ).
this routine for cursor moving, and by result-cache the fetch can be minimized.